### PR TITLE
Website: fix version display

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
This PR has two different aspects, one is "releasing" the 7.X version of the docs.
The other one is having a versions page so that when you click on the current version on the top left. Here is a video of this:

detox-website-versions

### CODE REVIEW

**Comment by rotemmiz** [comment]
I want to discuss doc duplication with you. I feel uncomfortable duping all docs 3 times (and potentially even more in the future). can't we generate the docs per version via git tags only on gh-pages branch during build ?

File: .travis.yml [hunk 1]
```diff
@@ -52,6 +52,12 @@ matrix:
 
     # Website
     - language: node_js
+      addons:
```
Comment (DanielMSchmidt) [inline .travis.yml]: This is needed by [nodegit](https://github.com/nodegit/nodegit)

**Comment by DanielMSchmidt** [comment]
@rotemmiz I thought you were not sure about storing them there, I didn't understand earlier that this is really a problem for you. 

The way I solved it now is by getting every git tag above version 6 (where we changed the docs to support docusaurus) and generate the version for each doc in a temporary copy of this repo, which is then copied over.

This is currently WIP, the sidebars don't work in the current implementation, need to figure out what's going wrong there 🤔 

Is this the general direction you want to go with this or did I misunderstand you? 😇

**Comment by DanielMSchmidt** [comment]
@rotemmiz This is done now / ready for review. 

I would like to move the code upstream to docusaurus, I think it is a better place for it. Easier to test and be concise this way (changing docusaurus versions could break this current workflow if they change something severe in the way they deal with versions). This is the issue I created for it: https://github.com/facebook/Docusaurus/issues/557

File: website/gatherDocs.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+const REPO_URL = "https://github.com/wix/detox.git"
+
+// From https://gist.github.com/joerx/3296d972735adc5b4ec1
+function clearRequireCache() {
+  Object.keys(require.cache).forEach(function (key) {
+    delete require.cache[key];
+  });
+}
+
+function getMajorVersion(tag) {
```
Comment (rotemmiz) [inline website/gatherDocs.js]: Please use 'semver' instead of parsing versions manually, this will break the pretty easily

Comment (DanielMSchmidt) [reply website/gatherDocs.js]: Good idea 👍

Comment (DanielMSchmidt) [reply website/gatherDocs.js]: I fixed that @rotemmiz 👍 Please review again!

**Comment by DanielMSchmidt** [comment]
@rotemmiz This is ready now 👍 I only clone the repo once and execute every action on this copy now :)

File: website/gatherDocs.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  await fs.emptyDir("./versioned_docs");
+  await fs.emptyDir("./versioned_sidebars");
+}
+
+async function checkoutVersion(repo, version) {
+  console.log("Checking out version", version);
+  await repo.checkoutBranch(version);
+}
+
+function applyTemporalFix(tempDir) {
```
Comment (rotemmiz) [inline website/gatherDocs.js]: `applyMissingDocHeadersForV6` or whatever version which needs it.

File: website/pages/en/versions.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
```
Comment (rotemmiz) [inline website/pages/en/versions.js]: Is this Facebook's code ?
If not, remove the license note

File: website/pages/en/versions.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+const CWD = process.cwd();
+
+const siteConfig = require(CWD + '/siteConfig.js');
+const versions = require(CWD + '/versions.json');
+const githubReleaseUrl = version => `https://github.com/wix/detox/releases/tag/${version}`;
+const documentationTarget = "Introduction.GettingStarted.html"
+
+function VersionLinks({version, isLatest = false}) {
+  const expandedVersion = version.replace('.X', '.0.0')
+  const isMaster = version === "master";
```
Comment (rotemmiz) [inline website/pages/en/versions.js]: Not sure I get what the final result will be. Will site show exact version (i.e. 7.3.5) or 7.0.0 throughout links ?

File: website/versioned_docs/version-6.X/APIRef.ActionsOnElement.md [hunk 6]
```diff
@@ -1,110 +0,0 @@
----
```
Comment (rotemmiz) [inline website/versioned_docs/version-6.X/APIRef.ActionsOnElement.md]: <3<3<3<3<3<3<3<3<3<3<3<3<3<3

File: website/pages/en/versions.js (hunk 5 above)
Comment (rotemmiz) [reply website/pages/en/versions.js]: Video shows 7.X, 6.X and master. what am I missing ?

File: website/pages/en/versions.js (hunk 4 above)
Comment (DanielMSchmidt) [reply website/pages/en/versions.js]: It's generated by docusaurus, should I remove this part?

File: website/pages/en/versions.js (hunk 5 above)
Comment (DanielMSchmidt) [reply website/pages/en/versions.js]: Video is old, but you can run it locally. Just do `cd website/ && npm install && npm run gatherDocs && npm start` and you can see the site

**Comment by DanielMSchmidt** [comment]
Damn it, the checkout seems to have failed, unnoticed by me. Need to do some more node-git magic here.

**Comment by DanielMSchmidt** [comment]
@rotemmiz I think I got your review comments, please take a look 👍

File: website/gatherDocs.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+async function cleanupExistingVersions() {
+  console.log("Cleanup versioned docs");
+  await fs.remove("./versions.json");
+  await fs.emptyDir("./versioned_docs");
+  await fs.emptyDir("./versioned_sidebars");
+}
+
+// https://stackoverflow.com/questions/29569913/switch-branch-tag-with-nodegit?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa
+function checkOutTag(repo, tag) {
```
Comment (rotemmiz) [inline website/gatherDocs.js]: LOL, utm_source will provide false data for trackers next time someone opens this link ;p
JK, but the query string can be removed

Comment (DanielMSchmidt) [reply website/gatherDocs.js]: Right, I'll get rid of that and see if I can get a better permalink

**Comment by DanielMSchmidt** [comment]
@rotemmiz Got a spare review?

**Pull request merged**